src/plugins/banner/public/services/banner_service_instance.ts

```diff
@@ -0,0 +1,4 @@
+import { BannerService } from './banner_service';
+
+// Create singleton instance of the banner service
```

Comment:
Singleton is not necessary. OSD’s plugin lifecycle itself manages service instances for you in a way that is functionally singleton-scoped within your plugin context. Each plugin in OSD is initialized once per process.The setup() and start() lifecycle methods are called once for each plugin instance.